odl/set/pspace.py

```diff
@@ -256,15 +258,16 @@ def element(self, inp=None):
         if inp is None:
             inp = [space.element() for space in self.spaces]
 
-        # TODO: how does this differ from "if inp in self"?
```

Comment:
Contains also verifies that the element itself (the tuple, if you will) is correct. This includes checks for the type of weights, inner product etc.